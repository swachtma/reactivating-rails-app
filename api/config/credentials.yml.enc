NruSqAfDHjCumPIyJn6xsY9LOZuE3ZzX2o/1QTdVkugVslYlIp4h/gmrnNPKDUKYZ8vGEGHPJ5EwlOSClaonGjgUB4yUG4tz8I0T++XLbiEU8qhxATlTf+s9JNfusXdgIB2ZFO4ABSDKhxhzPLbHYJVvLghx3WmwMMpO+GBgzlp94DLCX2IR38jeHEydFPy73oteodKVbTBp0g91v8DhpnZUTWHYb0JdqS+otrdrS1mar9MlcMnTpb8chvLisrrvE0SR/QUagAbd5UD/d02NczFYmAdKsyztbryVyakhgosZhCJFY3KNwxayxDC3sF5xFQHgli8qJNXNAhzaYDVf+n7D9k6g9BHTV5YRXdICgYORVWo+a9EhjBTmVwQsJN+IrdZaqxQVoa2DMEA3CVjOlp1wtLNwHbHArmId--8+ArDlMZvB3FEYTs--m4O75Mk62PJos0rR8zw91Q==