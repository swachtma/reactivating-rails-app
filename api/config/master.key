dc4eeb306d00d2a5e6993b94e449787e