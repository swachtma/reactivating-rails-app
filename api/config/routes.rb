Rails.application.routes.draw do
  scope :api do
    get 'github/', to: "authentication#github", format: false
    get 'hydrate_user/', to: "authentication#show"
    get 'nodes/', to: "nodes#index"
    get 'chapters/', to: "chapters#index"
    resources :bookmarks, only: [:index, :create]
  end
end