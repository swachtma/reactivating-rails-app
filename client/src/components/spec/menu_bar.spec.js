/* global expect, jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import config, { expectXInY } from '../../test_helper';
import MenuBar from '../menu_bar';


let nodes = ["#menu-bar","#side-menu-button","#side-menu-icon","Connect(UserBadge)","ChapterMenuItems"]
let mock_handler = jest.fn();
let w = shallow(<MenuBar handleSidebarVisibility={mock_handler} />);

describe("<MenuBar /> renders",()=>{
  nodes.forEach((n)=>{ it("a " + n, ()=>expectXInY(n,w)); });
  it("according to snapshot", () => expect(w).toMatchSnapshot());
});

describe("<MenuBar /> events", ()=>{
  it("calls handleSidebarVisibility onclick of #menu-bar", () =>{
    w.find("#menu-bar").simulate("click");
    expect(mock_handler).toBeCalled();
  });
});
