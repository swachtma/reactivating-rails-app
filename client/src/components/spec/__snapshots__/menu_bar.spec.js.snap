// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuBar /> renders according to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuBar
    handleSidebarVisibility={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "borderless": true,
      "children": <Container>
        <MenuItem
                fitted={true}
                id="side-menu-button"
                link={true}
                name="Main Menu"
        >
                <Icon
                        as="i"
                        color="black"
                        id="side-menu-icon"
                        name="content"
                        size="large"
                        style={
                                Object {
                                        "margin": "0",
                                      }
                        }
                />
        </MenuItem>
        <Image
                alt="Reactivating Rails"
                as="img"
                fluid={true}
                src="reactivating-rails.png"
                style={
                        Object {
                                "margin": "auto 1.5em",
                                "maxWidth": "230px",
                                "verticalAlign": "middle",
                              }
                }
                ui={true}
                wrapped={true}
        />
        <MenuMenu
                position="right"
                value={undefined}
        >
                <Responsive
                        getWidth={[Function]}
                        minWidth={768}
                >
                        <ChapterMenuItems
                                handleSidebarVisibility={[Function]}
                        />
                </Responsive>
                <Connect(UserBadge)
                        handleSidebarVisibility={[Function]}
                />
        </MenuMenu>
</Container>,
      "fixed": "top",
      "id": "menu-bar",
      "onClick": [Function],
      "size": "huge",
      "style": Object {
        "backgroundColor": "#fff",
        "border": "1px solid #ddd",
        "boxShadow": "0px 3px 5px rgba(0, 0, 0, 0.2)",
        "maxWidth": "100vw",
        "paddingLeft": "1em",
        "paddingRight": "1em",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MenuItem
            fitted={true}
            id="side-menu-button"
            link={true}
            name="Main Menu"
>
            <Icon
                        as="i"
                        color="black"
                        id="side-menu-icon"
                        name="content"
                        size="large"
                        style={
                                    Object {
                                                "margin": "0",
                                              }
                        }
            />
</MenuItem>,
          <Image
            alt="Reactivating Rails"
            as="img"
            fluid={true}
            src="reactivating-rails.png"
            style={
                        Object {
                                    "margin": "auto 1.5em",
                                    "maxWidth": "230px",
                                    "verticalAlign": "middle",
                                  }
            }
            ui={true}
            wrapped={true}
/>,
          <MenuMenu
            position="right"
            value={undefined}
>
            <Responsive
                        getWidth={[Function]}
                        minWidth={768}
            >
                        <ChapterMenuItems
                                    handleSidebarVisibility={[Function]}
                        />
            </Responsive>
            <Connect(UserBadge)
                        handleSidebarVisibility={[Function]}
            />
</MenuMenu>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Icon
              as="i"
              color="black"
              id="side-menu-icon"
              name="content"
              size="large"
              style={
                            Object {
                                          "margin": "0",
                                        }
              }
/>,
            "fitted": true,
            "id": "side-menu-button",
            "link": true,
            "name": "Main Menu",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "color": "black",
              "id": "side-menu-icon",
              "name": "content",
              "size": "large",
              "style": Object {
                "margin": "0",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "Reactivating Rails",
            "as": "img",
            "fluid": true,
            "src": "reactivating-rails.png",
            "style": Object {
              "margin": "auto 1.5em",
              "maxWidth": "230px",
              "verticalAlign": "middle",
            },
            "ui": true,
            "wrapped": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Responsive
                getWidth={[Function]}
                minWidth={768}
>
                <ChapterMenuItems
                                handleSidebarVisibility={[Function]}
                />
</Responsive>,
              <Connect(UserBadge)
                handleSidebarVisibility={[Function]}
/>,
            ],
            "position": "right",
            "value": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ChapterMenuItems
                  handleSidebarVisibility={[Function]}
/>,
                "getWidth": [Function],
                "minWidth": 768,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleSidebarVisibility": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleSidebarVisibility": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "borderless": true,
        "children": <Container>
          <MenuItem
                    fitted={true}
                    id="side-menu-button"
                    link={true}
                    name="Main Menu"
          >
                    <Icon
                              as="i"
                              color="black"
                              id="side-menu-icon"
                              name="content"
                              size="large"
                              style={
                                        Object {
                                                  "margin": "0",
                                                }
                              }
                    />
          </MenuItem>
          <Image
                    alt="Reactivating Rails"
                    as="img"
                    fluid={true}
                    src="reactivating-rails.png"
                    style={
                              Object {
                                        "margin": "auto 1.5em",
                                        "maxWidth": "230px",
                                        "verticalAlign": "middle",
                                      }
                    }
                    ui={true}
                    wrapped={true}
          />
          <MenuMenu
                    position="right"
                    value={undefined}
          >
                    <Responsive
                              getWidth={[Function]}
                              minWidth={768}
                    >
                              <ChapterMenuItems
                                        handleSidebarVisibility={[Function]}
                              />
                    </Responsive>
                    <Connect(UserBadge)
                              handleSidebarVisibility={[Function]}
                    />
          </MenuMenu>
</Container>,
        "fixed": "top",
        "id": "menu-bar",
        "onClick": [Function],
        "size": "huge",
        "style": Object {
          "backgroundColor": "#fff",
          "border": "1px solid #ddd",
          "boxShadow": "0px 3px 5px rgba(0, 0, 0, 0.2)",
          "maxWidth": "100vw",
          "paddingLeft": "1em",
          "paddingRight": "1em",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <MenuItem
              fitted={true}
              id="side-menu-button"
              link={true}
              name="Main Menu"
>
              <Icon
                            as="i"
                            color="black"
                            id="side-menu-icon"
                            name="content"
                            size="large"
                            style={
                                          Object {
                                                        "margin": "0",
                                                      }
                            }
              />
</MenuItem>,
            <Image
              alt="Reactivating Rails"
              as="img"
              fluid={true}
              src="reactivating-rails.png"
              style={
                            Object {
                                          "margin": "auto 1.5em",
                                          "maxWidth": "230px",
                                          "verticalAlign": "middle",
                                        }
              }
              ui={true}
              wrapped={true}
/>,
            <MenuMenu
              position="right"
              value={undefined}
>
              <Responsive
                            getWidth={[Function]}
                            minWidth={768}
              >
                            <ChapterMenuItems
                                          handleSidebarVisibility={[Function]}
                            />
              </Responsive>
              <Connect(UserBadge)
                            handleSidebarVisibility={[Function]}
              />
</MenuMenu>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Icon
                as="i"
                color="black"
                id="side-menu-icon"
                name="content"
                size="large"
                style={
                                Object {
                                                "margin": "0",
                                              }
                }
/>,
              "fitted": true,
              "id": "side-menu-button",
              "link": true,
              "name": "Main Menu",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "color": "black",
                "id": "side-menu-icon",
                "name": "content",
                "size": "large",
                "style": Object {
                  "margin": "0",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "Reactivating Rails",
              "as": "img",
              "fluid": true,
              "src": "reactivating-rails.png",
              "style": Object {
                "margin": "auto 1.5em",
                "maxWidth": "230px",
                "verticalAlign": "middle",
              },
              "ui": true,
              "wrapped": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Responsive
                  getWidth={[Function]}
                  minWidth={768}
>
                  <ChapterMenuItems
                                    handleSidebarVisibility={[Function]}
                  />
</Responsive>,
                <Connect(UserBadge)
                  handleSidebarVisibility={[Function]}
/>,
              ],
              "position": "right",
              "value": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <ChapterMenuItems
                    handleSidebarVisibility={[Function]}
/>,
                  "getWidth": [Function],
                  "minWidth": 768,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleSidebarVisibility": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "handleSidebarVisibility": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
